x-pack/plugins/enterprise_search/public/plugin.ts

```diff
@@ -52,6 +51,25 @@ export class EnterpriseSearchPlugin implements Plugin {
   }
 
   public setup(core: CoreSetup, plugins: PluginsSetup) {
+    core.application.register({
+      id: ENTERPRISE_SEARCH_PLUGIN.ID,
+      title: ENTERPRISE_SEARCH_PLUGIN.NAV_TITLE,
+      appRoute: ENTERPRISE_SEARCH_PLUGIN.URL,
+      category: DEFAULT_APP_CATEGORIES.enterpriseSearch,
+      mount: async (params: AppMountParameters) => {
+        const [coreStart] = await core.getStartServices();
+        const { chrome } = coreStart;
+        chrome.docTitle.change(ENTERPRISE_SEARCH_PLUGIN.NAME);
+
+        await this.getInitialData(coreStart.http);
+
+        const { renderApp } = await import('./applications');
+        const { EnterpriseSearch } = await import('./applications/enterprise_search');
+
+        return renderApp(EnterpriseSearch, params, coreStart, plugins, this.config, this.data);
+      },
+    });
```

Comment:
Super-NIT: @constancecchen I noticed we are registering a new app. Do you mind adding to [this file](https://github.com/elastic/kibana/blob/master/src/plugins/kibana_usage_collection/server/collectors/application_usage/schema.ts) an entry in the object `applicationUsageSchema` with the `ENTERPRISE_SEARCH_PLUGIN.ID` as a key?

We are working on an automated way of building that structure, but it has to be done manually for now. It can be done on a follow up PR, if you prefer to 🙂

Reply:
I can definitely do this - will push up in a separate commit here shortly!

Reply:
@afharo Please let me know if https://github.com/elastic/kibana/pull/76734/commits/6dee466b57dc31456ad363fc13c68ccded4eda5c looks correct to you! 🙇‍♀️